File: December 2023/Inorder Traversal and BST/main.py
# Link: https://www.geeksforgeeks.org/problems/inorder-traversal-and-bst5855/1


class Solution:
    def isRepresentingBST(self, arr, N):
        # code here
        stack = []
        prev_val = float("-inf")

        for val in arr:
            if val <= prev_val:
                return 0

            while stack and stack[-1] < val:
                prev_val = stack.pop()

            stack.append(val)

        return 1


if __name__ == "__main__":
    obj = Solution()
    arr = [2, 4, 3]
    n = len(arr)
    print(obj.isRepresentingBST(arr, n))

File: December 2023/Inorder Traversal and BST/main.cpp
// Link: https://www.geeksforgeeks.org/problems/inorder-traversal-and-bst5855/1

#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int isRepresentingBST(int arr[], int N)
    {
        // code here
        for (int i = N - 1; i > 0; i--)
            if (arr[i] <= arr[i - 1])
                return 0;
        return 1;
    }
};

int main()
{
    Solution obj;
    int N = 3;
    int arr[N] = {2, 4, 5};
    cout << obj.isRepresentingBST(arr, N);
    return 0;
}